components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,462 @@
+
+--
+-- match related functions
+--
+function matchFunctions.readDate(matchArgs)
+	return matchArgs.date
+		and MatchGroupInput.readDate(matchArgs.date)
+		or {
+			date = MatchGroupInput.getInexactDate(globalVars:get('tournament_enddate')),
+			dateexact = false,
+		}
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', '2v2'))
+	match.showh2h = Logic.emptyOr(match.showh2h, Variables.varDefault('showh2h'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+
+	-- apply vodgames
+	for index = 1, MAX_NUM_VODGAMES do
+		local vodgame = match['vodgame' .. index]
+		if not Logic.isEmpty(vodgame) then
+			local map = match['map' .. index] or {}
+			map.vod = map.vod or vodgame
+			match['map' .. index] = map
+		end
+	end
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	local opponent1 = match.opponent1 or {}
+	local opponent2 = match.opponent2 or {}
+
+	local casters = {}
+	for key, name in Table.iter.pairsByPrefix(match, 'caster') do
+		table.insert(casters, CustomMatchGroupInput._getCasterInformation(
+			name,
+			match[key .. 'flag'],
+			match[key .. 'name']
+		))
+	end
+	table.sort(casters, function(c1, c2) return c1.displayName:lower() < c2.displayName:lower() end)
+
+	local showh2h = Logic.readBool(match.showh2h)
+		and opponent1.type == Opponent.team
+		and opponent2.type == Opponent.team
+
+	match.extradata = {
+		showh2h = showh2h,
+		casters = Table.isNotEmpty(casters) and Json.stringify(casters) or nil,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getCasterInformation(name, flag, displayName)
+	if String.isEmpty(flag) then
+		flag = Variables.varDefault(name .. '_flag')
+	end
+	if String.isEmpty(displayName) then
+		displayName = Variables.varDefault(name .. '_dn')
+	end
+	if String.isEmpty(flag) or String.isEmpty(displayName) then
+		local parent = Variables.varDefault(
+			'tournament_parent',
+			mw.title.getCurrentTitle().text
+		)
+		local pageName = mw.ext.TeamLiquidIntegration.resolve_redirect(name)
+		local data = mw.ext.LiquipediaDB.lpdb('broadcasters', {
+			conditions = '[[page::' .. pageName .. ']] AND [[parent::' .. parent .. ']]',
+			query = 'flag, id',
+			limit = 1,
+		})
+		if type(data) == 'table' and data[1] then
+			flag = String.isNotEmpty(flag) and flag or data[1].flag
+			displayName = String.isNotEmpty(displayName) and displayName or data[1].id
+		end
+	end
+	if String.isNotEmpty(flag) then
+		Variables.varDefine(name .. '_flag', flag)
+	end
+	if String.isEmpty(displayName) then
+		displayName = name
+	end
+	if String.isNotEmpty(displayName) then
+		Variables.varDefine(name .. '_dn', displayName)
+	end
+	return {
+		name = name,
+		displayName = displayName,
+		flag = flag,
+	}
+end
+
+function matchFunctions.getOpponents(args)
```

Comment:
this function is pretty long imo it would be good to split it up a bit into several functions
would increase readability

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/b732c4b502a546d4644821284facf6cf8a20ba68